Sources/Controllers/2.0/OACustomPOIViewController.mm

```diff
@@ -161,79 +336,163 @@ - (void)selectSubcategoryDone:(OAPOICategory *)category keys:(NSMutableSet<NSStr
+            [tableView deselectRowAtIndexPath:indexPath animated:NO];
     }
-    return cell;
 }
 
 - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
 {
-    [tableView deselectRowAtIndexPath:indexPath animated:YES];
+    if (_searchMode)
+    {
+        [self selectDeselectItem:indexPath];
+    }
+    else
+    {
+        [tableView deselectRowAtIndexPath:indexPath animated:YES];
+
+        OAPOICategory* item = _categories[indexPath.row];
+        OASelectSubcategoryViewController *subcategoryScreen = [[OASelectSubcategoryViewController alloc] initWithCategory:item filter:_filter];
+        subcategoryScreen.delegate = self;
+        [self.navigationController pushViewController:subcategoryScreen animated:YES];
+    }
+}
+
+- (void)tableView:(UITableView *)tableView didDeselectRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    if (_searchMode)
+        [self selectDeselectItem:indexPath];
+}
 
-    OAPOICategory* item = _categories[indexPath.row];
-    OASelectSubcategoryViewController *subcategoryScreen = [[OASelectSubcategoryViewController alloc] initWithCategory:item filter:_filter];
-    subcategoryScreen.delegate = self;
-    [self.navigationController pushViewController:subcategoryScreen animated:YES];
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
+{
+    return 1;
 }
 
 - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
 {
-    return _categories.count;
+    return _searchMode ? _searchResult.count : _categories.count;
 }
 
-- (CGFloat)tableView:(UITableView *)tableView heightForFooterInSection:(NSInteger)section
+- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath
 {
-    return [OAPOISearchHelper getHeightForFooter];
+    return _searchMode;
 }
 
-- (CGFloat)tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section
+- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
 {
-    return [OAPOISearchHelper getHeightForHeader];
+    return _searchMode ? 48 : 66;
+}
+
+- (UIView *)tableView:(UITableView *)tableView viewForHeaderInSection:(NSInteger)section
+{
+    if (section == 0 && !_searchMode) {
+        OATableViewCustomHeaderView *customHeader = [tableView dequeueReusableHeaderFooterViewWithIdentifier:kHeaderId];
+        customHeader.label.text = [self getTitleForSection];
+        customHeader.label.font = [UIFont systemFontOfSize:15];
+        [customHeader setYOffset:20];
+        return customHeader;
+    }
+    return nil;
+}
+
+- (CGFloat)tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section {
+    if (section == 0 && !_searchMode) {
+        return [OATableViewCustomHeaderView getHeight:[self getTitleForSection] width:tableView.bounds.size.width] + 24;
+    }
+    return 0.01;
+}
+
+- (CGFloat)tableView:(UITableView *)tableView heightForFooterInSection:(NSInteger)section
```

Comment:
Set this in the xib and do not call this method. Also, please check other places OAPOISearchHelper is used, and remove this useless class from the project entirely.